columnphysics/icepack_atmo.F90

```diff
@@ -971,6 +984,10 @@ subroutine icepack_atm_boundary(sfctype,                   &
                                    uvel=l_uvel, vvel=l_vvel,  &
                                    Uref=l_Uref, zlvs=zlvs)
          if (icepack_warnings_aborted(subname)) return
+      else
+         call icepack_warnings_add( &
+            subname//' atmbndy = '//trim(atmbndy)//' : unknown value')
+         call icepack_warnings_setabort(.true.,__FILE__,__LINE__)
```

Comment:
...  and  the same thing here, maybe checking  for unknown values should also be in `icedrv_init` ? @apcraig @eclare108213 do we want to standardize on that for new code also in Icepack ?

Reply:
I raised this issue too.  We don't generally check "valid values" in Icepack.  There is quite a bit of that in CICE and the icepack driver.  However, users that incorporate Icepack in their own driver (like E3SM) will have to implement their own checking and won't be able to rely on Icepack to let them know if something is "off".  We might want to think about implementing explicit checks for Icepack arguments where it makes sense and is easy.  The would be a significant change but probably wouldn't change the model cost if done well (like checking values only when they are initialized/set or adding appropriate aborts in existing if logic).  But I think that is something we should discuss more broadly, create an issue if needed, and agree on general strategy before moving forward.

Reply:
I agree, no need to do this check here, or even in this PR.  In my opinion it would be better to keep the original behavior (just the 'else' without values) so that 'similarity' or 'mixed' (or 'default' / anything else) would all work.  Long ago, I treated default conditions in the code as exactly that -- it didn't matter what the flag was set to -- as long as it wasn't something specific (like 'constant') then the code would execute the default behavior.  Setting the flag to 'default' in the namelist file only made that explicit.  We seem to be moving to a more rigorous implementation, which is fine.  I prefer to do namelist value checking upon initialization, for better efficiency during the timestepping.  As @apcraig says, it's better for now to plan to address this value-checking issue more generally and not worry about it here.  The easiest thing to do now is keep the original behavior.

Reply:
Updated without checking `atmbndy` valid setting:  'constant' or 'mixed'. Anything else end in 'similarity' behavior.